language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/InterfaceClass.scala

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.daml.lf.codegen.backend.java.JavaEscaper
+import com.daml.lf.data.Ref.{ChoiceName, PackageId, QualifiedName}
+import com.daml.lf.iface.{DefInterface, TemplateChoice}
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+
+import javax.lang.model.element.Modifier
+
+object InterfaceClass extends StrictLogging {
+
+  def classNameForInterface(qualifiedName: QualifiedName) =
+    ClassName.bestGuess {
+      val QualifiedName(module, name) = qualifiedName
+      // consider all but the last name segment to be part of the java package name
+      val packageSegments = module.segments.slowAppend(name.segments).toSeq.dropRight(1)
+      // consider the last name segment to be the java class name
+      val className = name.segments.toSeq.takeRight(1)
+
+      val packageName = packageSegments.map(_.toLowerCase)
+
+      (packageName ++ className)
+        .filter(_.nonEmpty)
+        .map(JavaEscaper.escapeString)
+        .mkString(".")
```

Comment:
I see that you copied this from another place, but I believe there's a better way to do this. As a further note, this is possibly an instance of something that can be factored out in its own method (although I would stick to simple functions and not use inheritance).
```suggestion
      val packageName (module.segments.iterator ++ name.segments.iterator).init.map(_.toLowerCase)
      // consider the last name segment to be the java class name
      val className = Iterator.single(name.segments.last)

      (packageName ++ className)
        .filter(_.nonEmpty)
        .map(JavaEscaper.escapeString)
        .mkString(".")
```
`ImmArray.last` is `O(1)` (should be documented alongside the method) and using iterators we can avoid materializing the collection and copying stuff around unnecessarily. Please do double check that my code compiles, I wrote it from within Github.

Reply:
I refactored it such that the initial function this was coming from is being used instead of code duplication